### DISCUSSION TITLE
nuxt middleware, after refresh should access to browser api?

### DISCUSSION THREAD

**Discussion by hackpirodev** [nuxt/nuxt#24006] [Questions]
Using supabase I notice that my middleware doesn't work as expected after refresh. Then I tried to create a minimal case and I discovered that nuxt middleware can't access to browser API. 

If I navigate normally, middleware can access for example localStorage. But if I refresh the page where I have my middleware active nuxt gives me a 500 error because has no localStorage.

Is this behaviour normal or should I open an issue?

![image](https://github.com/nuxt/nuxt/assets/68652588/e9d33358-9777-4a05-9b0f-270576b0882f)

here minimal case 
https://github.com/hackpirodev/nuxt-middleware-async

**Comment by xjccc** [comment]
use `process.client` to resolve this?

because `localStorage` is use in client
